corehq/apps/hqcase/tests/test_case_api_permissions.py

```diff
@@ -0,0 +1,213 @@
+import uuid
+from contextlib import contextmanager
+
+from django.test import TestCase
+from django.urls import reverse
+
+from casexml.apps.case.mock import CaseBlock
+
+from corehq import privileges
+from corehq.apps.commtrack.tests.util import bootstrap_domain
+from corehq.apps.es.tests.utils import (
+    case_search_es_setup,
+    case_search_es_teardown,
+    es_test,
+)
+from corehq.apps.hqcase.api.get_list import get_list
+from corehq.apps.locations.tests.util import setup_locations_and_types
+from corehq.apps.users.models import HqPermissions, UserRole, WebUser
+from corehq.elastic import get_es_new
+from corehq.pillows.mappings.user_mapping import USER_INDEX
+from corehq.util.elastic import ensure_index_deleted
+from corehq.util.es.testing import sync_users_to_es
+from corehq.util.test_utils import (
+    disable_quickcache,
+    flag_enabled,
+    privilege_enabled,
+)
+
+
+@es_test
```

Comment:
You can use the `es_test` decorator to specify which index(es) your tests need. It will perform all the setup/teardown index boilerplate for you.

Further note: It looks like `test_get_list_location_restricted()` is the only test that actually needs Elasticsearch. If that's true, you can move this `es_test` decorator to decorate only that method. If other tests in this case also need Elasticsearch though, then this is the correct place for the decorator.

```suggestion
@es_test(requires=[user_adapter])
```

Reply:
It would be nice to make Elasticsearch would work like a SQL db during tests: each test is performed inside a (pseudo, for ES) transaction that gets rolled back when the test/Case/cls context exits. This is possible in theory, and I don't think it would be hard to implement (although it might be challenging to implement _efficiently_).